lib/matplotlib/testing/decorators.py

```diff
@@ -504,3 +504,62 @@ def _image_directories(func):
     result_dir = Path().resolve() / "result_images" / module_path.stem
     result_dir.mkdir(parents=True, exist_ok=True)
     return baseline_dir, result_dir
+
+
+def generate_matplotlib_baseline_images_from_previous_commit():
+    mplRepoPath = Path(mpl.__file__).parent
+    # Check that git is installed.
+    subprocess.run(shutil.which('git'),
+                    stderr=subprocess.STDOUT,
+                    stdout = open(os.devnull, 'w'),
+                    universal_newlines=True) == None
+    # Check if current repository is a git repository
+    subprocess.run(['git', 'status'],
+                    cwd = mplRepoPath,
+                    stderr=subprocess.STDOUT,
+                    stdout = open(os.devnull, 'w'),
+                    universal_newlines=True) == None
+    # Create virtual env in tmpdir.
+#    if not (mplRepoPath / 'tmp/testenv').exists():
+#        modify_existing_baseline_images()
+#    else
+        subprocess.run(["python", "-mvenv", "tmp/testenv"],
+                        cwd = mplRepoPath,
+                        stderr=subprocess.STDOUT,
+                        stdout=subprocess.DEVNULL),
+                        universal_newlines=True)
+        # Clone mpl repo to tmpvenv and run pytests from new mpl repo created
+        subprocess.run(["git", "clone", str(mplRepoPath), "tmp/testenv"],
+                        cwd = mplRepoPath,
+                        stderr=subprocess.STDOUT,
+                        stdout=subprocess.DEVNULL,
+                        universal_newlines=True)
+        subprocess.run(["tmp/testenv/bin/python"],
+                        cwd = mplRepoPath,
+                        stderr=subprocess.STDOUT,
+                        stdout=subprocess.DEVNULL,
+                        universal_newlines=True)
+        subprocess.run(["git", "checkout", "HEAD~"],
+                        cwd = mplRepoPath,
+                        stderr=subprocess.STDOUT,
+                        stdout=subprocess.DEVNULL,
+                        universal_newlines=True)
+        subprocess.run(["python", "-mpip", "install", "-e", mplRepoPath],
+                        cwd = mplRepoPath,
+                        stderr=subprocess.STDOUT,
+                        stdout=subprocess.DEVNULL,
+                        universal_newlines=True)
+        subprocess.run(["python", "-mpytest"],
```

Comment:
Should also install pytest and all of the extra dependencies (pandas, ...) to make sure we generate all of the images or should we only match what the user has installed in their development environment?

Reply:
I think installing all extra dependencies is a better option as testing aims to check integrity amongst the components too, which can't be achieved if only some of the installs are done. 
First time all the images will be generated, later on we can move on specific missing installs and modifications of the baseline images.

Reply:
What do you think ?